src/main/kotlin/eu/europa/ec/eudi/pidissuer/port/input/GetTypeMetadata.kt

```
@@ -0,0 +1,69 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package eu.europa.ec.eudi.pidissuer.port.input
+
+import arrow.core.Either
+import arrow.core.raise.Raise
+import arrow.core.raise.catch
+import arrow.core.raise.either
+import eu.europa.ec.eudi.sdjwt.vc.SdJwtVcTypeMetadata
+import eu.europa.ec.eudi.sdjwt.vc.Vct
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import org.springframework.core.io.ClassPathResource
+import java.nio.charset.StandardCharsets
+
+// Sealed interface may be not needed here and subject to removal
+sealed interface MetadataSuccessResponse {
+    @Serializable
+    data class Metadata(
+        val metadata: SdJwtVcTypeMetadata,
+    ) : MetadataSuccessResponse
+}
+
+@Serializable
+data class GetTypeMetadataError(val error: String) {
+    companion object {
+        val InvalidVct = GetTypeMetadataError("Provided string could not be converted to VCT")
+        val UnrecognisedVct = GetTypeMetadataError("Domain does not recognise VCT")
+        val MetadataIncorrectFormat = GetTypeMetadataError("Metadata incorrect format")
+    }
+}
+
+class GetTypeMetadata(
+    private val knownVct: Map<Vct, ClassPathResource>,
+) {
+    operator fun invoke(vct: String): Either<GetTypeMetadataError, MetadataSuccessResponse> = either {
+        val vct = convertToVct(vct)
+        val knownClassPath = knownVct.getOrElse(vct) { raise(GetTypeMetadataError.UnrecognisedVct) }
+        val convertedMetadata: SdJwtVcTypeMetadata = convertMetadataToSdJwtVcTypeMetadata(knownClassPath)
+
+        MetadataSuccessResponse.Metadata(convertedMetadata)
+    }
+
+    fun Raise<GetTypeMetadataError>.convertToVct(vct: String): Vct =
+        catch({ Vct(vct) }) {
+            raise(GetTypeMetadataError.InvalidVct)
+        }
+
+    fun Raise<GetTypeMetadataError>.convertMetadataToSdJwtVcTypeMetadata(knownClassPath: ClassPathResource): SdJwtVcTypeMetadata =
```

@atsant No need for this specific error. let it fail. if the json is not correct, it's our fault. not something to report to the user.